darshan-util/pydarshan/darshan/experimental/plots/heatmap_handling.py

```diff
@@ -273,7 +273,7 @@ def get_aggregate_data(
     return agg_df
 
 
-def get_heatmap_df(agg_df: pd.DataFrame, xbins: int, nprocs: int) -> pd.DataFrame:
+def get_heatmap_df(agg_df: pd.DataFrame, xbins: int, nprocs: int, tmax: float = None) -> pd.DataFrame:
```

Comment:
I'm not sure there's much value in changing `get_heatmap_df` in this PR--there are no new tests for the new functionality and we aren't using it in this PR?

We probably also want the new argument typed as `Optional[float]` to avoid the potential future issue with:

`mypy --no-implicit-optional darshan/experimental/plots/heatmap_handling.py `

`darshan/experimental/plots/heatmap_handling.py:276: error: Incompatible default for argument "tmax" (default has type "None", argument has type "float")  [assignment]`

Though I'd probably just leave `get_heatmap_df()` alone in this PR at this point.